mongo/mongo-derivation/src/main/scala/io/sphere/mongo/generic/MongoFormatMacros.scala

```diff
@@ -107,8 +108,12 @@ private[generic] object MongoFormatMacros {
         )
       else {
         val subtypes = collectKnownSubtypes(c)(symbol)
-        val subtypesWithNoFormatter = subtypes.filterNot(mongoFormatExists(c))
-        val idents = Ident(symbol.name) :: subtypes.map { s =>
+
+        val (subtypesWithFormatter, subtypesWithNoFormatter) =
+          subtypes.partition(mongoFormatExists(c))
+        val singleParamTypes = subtypesWithFormatter.filter(_.asType.typeParams.length == 1)
+
+        val idents = Ident(symbol.name) :: (subtypes -- singleParamTypes).map { s =>
```

Comment:
We generate custom TypeSelectors for types with type parameters, so we remove them from this list. (The idents list will be the type parameter list of the mongoTypeSwitch method)